packages/lesswrong/server/callbacks/postCallbacks.ts

```diff
@@ -520,3 +525,63 @@ getCollectionHooks("Posts").updateAfter.add(async (post: DbPost, props: CreateCa
 
   return post;
 });
+
+/**
+ * Call the Type3 webhook to notify it of a new post. This will trigger audio to be pre-generated for the post.
+ */
+async function callType3Webhook(action: 'post_published' | 'post_edited', url: string) {
+  const clientId = type3ClientIdSetting.get();
+  const webhookSecret = type3WebhookSecretSetting.get();
+
+  if (!clientId || !webhookSecret) return;
+
+  const webhookUrl = 'https://api.type3.audio/webhooks/create/';
+  const data = {
+    client_id: clientId,
+    action: action,
+    url: url,
+    key: webhookSecret,
+  };
+
+  const res = await fetch(webhookUrl, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(data),
+  });
+
+  if (res.status !== 200) {
+    // eslint-disable-next-line no-console
+    console.log(`Failed to call Type3 webhook with action ${action} for post ${url}. Response code: `, res.status)
+  }
+}
+
+/**
+ * Call the Type 3 webhook on post creation
+ */
+getCollectionHooks("Posts").createAsync.add(async ({document}: CreateCallbackProperties<DbPost>) => {
+  if (!document.authorIsUnreviewed && !document.draft) {
+    const url = postGetPageUrl(document);
+    await callType3Webhook('post_published', url);
```

Comment:
Do we actually need to `await` these callbacks or should we just `void` them? I'm slightly worried that they could delay other important things if the hooks get added after these ones and the API is a little slow.

Reply:
Good point, I've changed them to `void`